sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -863,14 +863,79 @@ def delete_blob(
             The timeout parameter is expressed in seconds.
         :rtype: None
         """
-        blob = self.get_blob_client(blob) # type: ignore
+        blob_client = self.get_blob_client(blob) # type: ignore
         kwargs.setdefault('merge_span', True)
         timeout = kwargs.pop('timeout', None)
-        blob.delete_blob( # type: ignore
+        blob_client.delete_blob( # type: ignore
             delete_snapshots=delete_snapshots,
             timeout=timeout,
             **kwargs)
 
+    @distributed_trace
+    def download_blob(self, blob, offset=None, length=None, **kwargs):
+        # type: (Union[str, BlobProperties], Optional[int], Optional[int], **Any) -> StorageStreamDownloader
+        """Downloads a blob to a stream with automatic chunking.
```

Comment:
probably we want to improve the doc to avoid misleading the users? 
Actually this method is not download a blob into stream. It's download the blob partially into StorageStreamDownloader, and user need to use StorageStreamDownloader to download the blob content by calling readall() or download the blob into a stream by calling readinto().

Reply:
Thats a good point! We need to improve the doc here now that we have chance :)